01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+
+Initially written by Jung Hyun Choi and updated by Manu Alcalá Kovalski in 2023-2024.
+
+## Process
+
+-   [Housekeeping](#housekeeping)
+-   [Read Data](#read-data)
+-   [Clean Data](#clean-data)
+-   [Merge Crosswalk](#merge-crosswalk)
+-   [Create Housing Value Metric](#create-metric)
+-   [Evaluate Results](#evaluate-results)
+-   [Export](#export)
+
+## Housekeeping {#sec-housekeeping}
+
+```{r}
+#| label: setup
+
+options(scipen = 999)
+librarian::shelf(tidyverse, ipumsr, here, glue, labelled, skimr, urbnthemes, tidylog, assertr)
+set_urbn_defaults(style = "print")
+```
+
+### Read Data {#sec-read-data}
+
+We extract the 1 year ACS files from the IPUMS API using `ipumsr`. You will need to have an API key in order to do so. If you don't have one, you can register to the [IPUMS USA](https://uma.pop.umn.edu/usa/user/new?return_url=https%3A%2F%2Fusa.ipums.org%2Fusa-action%2Fmenu) data collection and create an API key.
+
+By default, `ipumsr` API functions assume that your key is stored in the **IPUMS_API_KEY** environment variable. You can also provide your key directly to these functions, but storing it in an environment variable saves you some typing and helps prevent you from inadvertently sharing your key with others (for instance, on GitHub).
+
+You can save your API key to the IPUMS_API_KEY environment variable with `set_ipums_api_key(api_key = "paste-your-key-here", save = TRUE)`. This will add your API key to your .Renviron file in your user home directory.
+
+```{r}
+#| label: query-ipums-api
+#| cache: true
+#| eval: false
+## 2.1. ACS: keep households living not living in gq
+
+# Run this the first time you run this code to avoid requesting the same data
+# multiple times
+
+acs_vars <- c("YEAR", "SAMPLE", "SERIAL", "CBSERIAL", "HHWT", "CLUSTER", "state", "county", "MET2013", "CITY", "PUMA", "STRATA", "GQ", "OWNERSHP", "OWNERSHPD", "VALUEH", "PERNUM", "PERWT", "RACE", "RACED", "HISPAN", "HISPAND", "AGE")
+
+years <- 2014:2022
+# usa_ext_def <- define_extract_usa(
+#   description = "ACS 1 Year 2014-2022",
+#   samples = paste0("us", years, "a"),
+#   variables = acs_vars
+# )
+#
+# usa_ext_submitted <- submit_extract(usa_ext_def)
+# usa_ext_complete <- wait_for_extract(usa_ext_submitted)
+#
+# # download_extract() will return a character vector including the file paths
+# # to all downloaded files.
+#
+# filepath <- download_extract(usa_ext_complete,
+#   download_dir = here("01_financial-well-being/data")
+# )
+```
+
+### Clean data {#sec-clean-data}
+
+We load the 2014-2022 1 Year ACS microdata we downloaded from IPUMS above. IPUMS loads the data as a person-level data set with some household variables.Therefore, we can analyze the data at the household level by keeping the first person sampled in the household.
+
+```{r}
+#| label: load-acs
+#| cache: true
+#| dependson: "import-acs-one-year"
+#| include: false
+
+# Set microdata to whatever the file you downloaded in the previous chunk is called
+microdata <- "usa_00010.xml"
+
```

Comment:
I would add a super-clear note in the text above this code that line XX (currently 97) would need to be manually changed to the extract number the analyst gets from the code.

Reply:
Updated the comment accordingly!

Reply:
Perfect!